pkg/client/group_mgmt_client.go

```diff
@@ -69,19 +71,31 @@ func newGroupMgmtClient(ipAddress, username, password, apiVersion string, waitOn
 
 	// Create GroupMgmt Client
 	groupMgmtClient := &GroupMgmtClient{
-		URL:       fmt.Sprintf("https://%s:5392/%s", ipAddress, apiVersion),
 		Client:    restyClient,
 		WaitOnJob: waitOnJobs,
 		Username:  username,
 		Password:  password,
+		isTenant:  isTenant,
 	}
+
+	if groupMgmtClient.isTenant {
+		groupMgmtClient.URL = fmt.Sprintf(groupURIFmtTenant, ipAddress, apiVersion)
+	} else {
+		groupMgmtClient.URL = fmt.Sprintf(groupURIFmt, ipAddress, apiVersion)
+	}
+
 	return groupMgmtClient
 }
 
 // NewClient instantiates a new client to communicate with the Nimble group
-func NewClient(ipAddress, username, password, apiVersion string, waitOnJobs bool) (*GroupMgmtClient, error) {
+func NewClient(ipAddress, username, password, apiVersion string, waitOnJobs, isTenant bool) (*GroupMgmtClient, error) {
 	// Get resty client
-	groupMgmtClient := newGroupMgmtClient(ipAddress, username, password, apiVersion, waitOnJobs)
+	groupMgmtClient := newGroupMgmtClient(ipAddress, username, password, apiVersion, waitOnJobs, isTenant)
+
+	if apiVersion != "v1" {
+		return nil, fmt.Errorf("NewNimbleGroupService: unsupported %s sdk API version", apiVersion)
```

Comment:
Remove "NewNimbleGroupService".  I suggest:

    return nil, fmt.Errorf("API version \"%s\" is not recognized", apiVersion)

Reply:
Now that this function is moved to newGroupMgmtClient(), it can not return the error message (because newGroupMgmtClient only return 1 value that is *GroupMgmtClient). Should I just do  this instead:
```
fmt.Println("API version \"%s\" is not recognized", apiVersion)
return nil
```

Reply:
Why is the check moved to newGroupMgmtClient()?  If necessary, we can add an error return value to that function.  Println is not a good option.

Reply:
Oh, never mind. I misunderstood your other comment from last night. I'm gonna keep the check over here.

Reply:
Is another commit incoming for this one?